### ISSUE TITLE
using chin movement to determine that user is still talking

### ISSUE THREAD

**Issue by jmossbridge** [hansonrobotics/hr_reflexes#6]
Right now, once Sophia determines that the user has stopped talking, she starts talking and the mic is shut off, resulting in a feeling of being interrupted if the user  has just paused a bit. One solution is to use chin movement in the user (instead of mic input) after the mic has shut off to determine  if the user is still speaking, then have Sophia do a brief, "I'm sorry, what were you saying?" to cue the person to repeat.

**Comment by elggem** [comment]
Yes, that would be great! We don't have a clear idea yet who amongst all the faces is the active speaker, but this is only an issue for multi speaker settings. Let's disregard it for now and build it for the single-speaker scenario. 

Of course, there is some interaction from sensory input to Chatscript involved here to make her go back in the dialog tree to before the last prompt, but I will look into the necessary parts of the pipeline to see if this is possible / how much effort it will be without Ghost!

I think there is a trigger somewhere (used by the shut-up command) to make her shut-up/pause at least!

More later...

**Comment by elggem** [comment]
So the first implementation of this is online here:

https://github.com/hansonrobotics/hr_reflexes/blob/master/scripts/interruption.py

It seems to work reasonably well, will do more testing and tuning in the next few days.

Still missing is:

  - [ ] Have an array of different apologies so it doesnt always sound the same. 
  - [ ] Backtrack in the conversation tree, which @eddiemonroe has told me can theoretically be done with `:retry` or `:redo`… Unfortunately this doesnt seem to be working due to a CS bug… I will ask @wenwei-dev  to pull the latest release (or tell me how to) then I can test it.
